knx/knxnet/search.go

```diff
@@ -0,0 +1,84 @@
+// Licensed under the MIT license which can be found in the LICENSE file.
+
+package knxnet
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/vapourismo/knx-go/knx/util"
+)
+
+// A SearchReq requests a discovery from all KNXnet/IP Servers
+type SearchReq struct {
+	HostInfo
+}
+
+// Service returns the service identifier for search request
+func (SearchReq) Service() ServiceID {
+	return SearchReqService
+}
+
+// SetHostInfo sets the address where ObjectServers should send the reponse to
+func (req *SearchReq) SetHostInfo(addr net.Addr) error {
```

Comment:
Could this be a function `NewHostInfo` returning `(*SearchReq, error)` instead?

Reply:
Of course, I can change that.

Reply:
Done